.github/workflows/tests.yml

```diff
@@ -20,218 +8,100 @@ jobs:
+          # Ubuntu with the oldest supported GCC
+          # (used to be 4.8, but with 20.04 it is 7, sadly).
+          - {os: ubuntu-20.04, gcc: 7, python: 3.8}
+          - {os: ubuntu-20.04, gcc: 11, python: 3.9}
+          # macOS, oldest supported version
+          # (macos-10.15 and macos-11 were removed by GitHub)
+          - {os: macos-12, python: 3.8}
           # macOS, newest supported version (haven't tried macos-14 yet)
-          - os: macos-13
-            python: 3.9
-
+          - {os: macos-13, python: 3.9}
+      
+    runs-on: ${{ matrix.os }}
     steps:
-      - name: Checkout
-        uses: actions/checkout@v2
-
-      - name: Choose GCC
-        if: startsWith(matrix.os, 'ubuntu')
-        run: |
-          # Install specific GCC version as the default.
-          sudo apt-get install gcc-${{ matrix.gcc }} g++-${{ matrix.gcc }}
-          sudo update-alternatives \
-            --install /usr/bin/gcc gcc /usr/bin/gcc-${{ matrix.gcc }} 100 \
-            --slave /usr/bin/g++ g++ /usr/bin/g++-${{ matrix.gcc }}
-
-      - name: Cache conda and dependancies
-        id: cache
-        uses: actions/cache@v2
-        with:
-          path: |
-            c:\Miniconda\envs\anaconda-client-env
-            /usr/share/miniconda/envs/anaconda-client-env
-            ~/osx-conda
-            ~/.profile
-          key: ${{ runner.os }}-${{ matrix.python }}-conda-v13-${{ hashFiles('treerec/tests/conda-requirements.txt') }}-${{ hashFiles('treerec/tests/pip-requirements.txt') }}
-
-      - name: Install Conda
-        uses: conda-incubator/setup-miniconda@v2
-        if: steps.cache.outputs.cache-hit != 'true'
+      - name: Check out repository code
+        uses: actions/checkout@v4
+      - name: Setup Mambaforge
+        uses: conda-incubator/setup-miniconda@v3
         with:
+          miniforge-variant: Mambaforge
+          miniforge-version: latest
           activate-environment: anaconda-client-env
+          use-mamba: true
           python-version: ${{ matrix.python }}
-          channels: conda-forge
-          channel-priority: strict
-          auto-update-conda: true
-          use-only-tar-bz2: true
-
-      - name: Install conda deps
-        if: steps.cache.outputs.cache-hit != 'true'
-        shell: bash -l {0}    # We need a login shell to get conda
-        run: conda install --yes --file=treerec/tests/conda-requirements.txt
-
-      - name: Install pip deps
+      - name: Get Date
+        id: get-date
+        run: echo "today=$(/bin/date -u '+%Y%m%d')" >> $GITHUB_OUTPUT
+        shell: bash
+      - name: Cache Conda env
+        uses: actions/cache@v3
+        with:
+          path: ${{ env.CONDA }}/envs
+          key:
+            conda-${{ runner.os }}--python-${{ matrix.python }}--${{ runner.arch }}--${{
+            steps.get-date.outputs.today }}-${{
+            hashFiles('treerec/tests/environment.yml') }}-${{ env.CACHE_NUMBER}}
+        env:
+          # Increase this value to reset cache if
```

Comment:
Modifying the value of the CACHE_NUMBER  would trigger the creation of a new environment and marked the previous one as "obsolete"

Reply:
OK; we just change it to, e.g., 1, and that triggers a cache rebuild; and we just leave it at 1, and perhaps eventually move it to 2, etc., there's no need to ever move it back down to 0.

Reply:
I think that is the idea :)